cli/src/commands/parallelize.rs

```diff
@@ -0,0 +1,192 @@
+        .collect();
+    if target_commits.len() < 2 {
+        return Ok(writeln!(ui.status(), "Nothing changed.")?);
+    }
+    let target_revset = RevsetExpression::commits(target_commits.clone());
+    let target_head = get_head(&target_revset, tx.repo())?;
+    let target_root = get_root(&target_revset, tx.repo())?;
+    let connected_length = RevsetExpression::commits(vec![target_head, target_root.clone()])
+        .connected()
+        .evaluate_programmatic(tx.repo())?
+        .iter()
+        .count();
+    if connected_length != target_commits.len() {
+        return Err(user_error(
+            "Cannot parallelize since the target revisions are not connected.",
+        ));
+    }
+
+    // Rebase the children of each target commit onto its new parents. A child
+    // which had a target commit as an ancestor before parallelize ran will have
+    // the target commit as a parent afterward.
+    for target_commit in target_commits.iter() {
+        // These parents are shared by all children of the target commit and
+        // include the target commit itself plus any of its ancestors which are
+        // being parallelized.
+        let common_parents: IndexSet<CommitId> = target_revset
+            .intersection(&RevsetExpression::commit(target_commit.clone()).ancestors())
+            .evaluate_programmatic(tx.repo())?
+            .iter()
+            // We want parents before children, so we need to reverse the order.
+            .reversed()
+            .collect();
+        // Children of the target commit, excluding other target commits.
+        let children: Vec<Commit> = RevsetExpression::commit(target_commit.clone())
+            .children()
+            .minus(&target_revset)
+            .evaluate_programmatic(tx.repo())?
+            .iter()
+            .commits(tx.repo().store())
+            .try_collect()?;
+        for child in children {
+            let mut new_parents = common_parents.clone();
+            new_parents.extend(child.parent_ids().iter().cloned());
+            let parents = new_parents
+                .iter()
+                .map(|c| tx.repo().store().get_commit(c))
+                .try_collect::<_, Vec<Commit>, _>()?;
+            rebase_descendants(
+                &mut tx,
+                command.settings(),
+                &parents,
+                &[child],
+                Default::default(),
+            )?;
+        }
+    }
+
+    // Rebase the target commits onto the parents of the root commit.
+    let new_parents = tx.repo().store().get_commit(&target_root)?.parents();
+    let target_commits = target_commits
+        .iter()
+        .map(|c| tx.repo().store().get_commit(c))
+        .try_collect::<_, Vec<Commit>, _>()?;
+    rebase_descendants(
+        &mut tx,
+        command.settings(),
+        &new_parents,
+        &target_commits,
+        Default::default(),
+    )?;
+
+    tx.finish(
+        ui,
+        format!("Parallelized {} commits.", target_commits.len()),
```

Comment:
nit: "parallelize {} commits"

op log message usually starts with lowercase letter, is present tense, has no trailing period.

Reply:
That's actually bothered me a bit for a while. I can send a PR to make it uppercase and include a trailing period. That would better match the style in our terminal messages. Let me know if you disagree.

Reply:
> make it uppercase and include a trailing period.

I don't mind about uppercase/lowercase thing, but op log description is similar to a commit message, which doesn't usually contain trailing period in the summary line.